### ISSUE TITLE
Swarm: http errors due to trailing slash 

### ISSUE THREAD

**Issue by homotopycolimit** [ethereum/go-ethereum#2886]
TL;DR http://swarm-gateways.net/bzz:/swarm/ works but http://swarm-gateways.net/bzz:/swarm does not.

The issue with the trailing slash in swarm.

Problem: The swarm landing page works when called like this
http://52.169.106.34:32208/bzz:/49d31de7f834a2d8b9d7e4a8f365ba9b5df8b1215c4f13cae4a7179797155f2b/
or this:
http://swarm-gateways.net/bzz:/swarm/

but fails when called like this:
http://52.169.106.34:32208/bzz:/49d31de7f834a2d8b9d7e4a8f365ba9b5df8b1215c4f13cae4a7179797155f2b
or this:
http://swarm-gateways.net/bzz:/swarm

the reason is that while the name 'swarm' or the hash "49d31de7f834..." return the same html content regardless of whether you include a trailing /, the realative paths within the html document are interptreted differently. In one case a realtive path ./tohere will reference a file subrdinate to 'swarm' -- that is bzz:/swarm/tohere -- whereas in the other case it will reference a (non-existing) file bzz:/tohere

The slash "/" is a special character for the browser even if it is deliberately not special in swarm.

I see no easy way of fixing this issue because it seems to be based on an incompatibility between how browsers treat the / character and how they are handled in manifests.

Follow up question: can I register the name "swarm/" too and what happens then when I call swarm-gateways.net/bzz:/swarm/

**Comment by zelig** [comment]
fixed in https://github.com/ethereum/go-ethereum/pull/2899
